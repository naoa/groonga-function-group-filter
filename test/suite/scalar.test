plugin_register functions/group_filter
table_create applicants TABLE_HASH_KEY ShortText
table_create ipcs TABLE_HASH_KEY ShortText
table_create docs TABLE_HASH_KEY ShortText
column_create docs applicant COLUMN_SCALAR applicants
column_create docs ipcs COLUMN_VECTOR ShortText
load --table docs
[
{"_key": "JP2001213456", "applicant": "三洋", "ipcs": ["G06F", "G04A"]},
{"_key": "JP2001213457", "applicant": "パナソニック", "ipcs": ["G04A"]},
{"_key": "JP2001213458", "applicant": "ソニー", "ipcs": ["G06F", "G04B"]},
{"_key": "JP2001213459", "applicant": "パナソニック", "ipcs": ["G06F"]}
]
column_create applicants applicant COLUMN_INDEX docs applicant
column_create ipcs ipcs COLUMN_INDEX docs ipcs

select docs \
  --filter 'group_filter(applicant, 2)' \
  --output_columns _id,_score,_key,applicant,#group_applicant
