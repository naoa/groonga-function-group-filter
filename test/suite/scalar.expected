plugin_register functions/group_filter
[[0,0.0,0.0],true]
table_create applicants TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
table_create ipcs TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
table_create docs TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
column_create docs applicant COLUMN_SCALAR applicants
[[0,0.0,0.0],true]
column_create docs ipcs COLUMN_VECTOR ShortText
[[0,0.0,0.0],true]
load --table docs
[
{"_key": "JP2001213456", "applicant": "三洋", "ipcs": ["G06F", "G04A"]},
{"_key": "JP2001213457", "applicant": "パナソニック", "ipcs": ["G04A"]},
{"_key": "JP2001213458", "applicant": "ソニー", "ipcs": ["G06F", "G04B"]},
{"_key": "JP2001213459", "applicant": "パナソニック", "ipcs": ["G06F"]}
]
[[0,0.0,0.0],4]
column_create applicants applicant COLUMN_INDEX docs applicant
[[0,0.0,0.0],true]
column_create ipcs ipcs COLUMN_INDEX docs ipcs
[[0,0.0,0.0],true]
select docs   --filter 'group_filter(applicant, 2)'   --output_columns _id,_score,_key,applicant,#group_applicant
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        3
      ],
      [
        [
          "_id",
          "UInt32"
        ],
        [
          "_score",
          "Int32"
        ],
        [
          "_key",
          "ShortText"
        ],
        [
          "applicant",
          "applicants"
        ],
        [
          "#group_applicant",
          "applicants"
        ]
      ],
      [
        2,
        1,
        "JP2001213457",
        "パナソニック",
        [
          "パナソニック"
        ]
      ],
      [
        4,
        1,
        "JP2001213459",
        "パナソニック",
        [
          "パナソニック"
        ]
      ],
      [
        3,
        1,
        "JP2001213458",
        "ソニー",
        [
          "ソニー"
        ]
      ]
    ]
  ]
]
